examples/erc721/src/lib.rs

```diff
@@ -0,0 +1,810 @@
+    logger: &mut impl HasLogger,
+    state: &mut State,
+) -> ContractResult<A> {
+    let params: ApproveParams = ctx.parameter_cursor().get()?;
+    let sender = ctx.sender();
+    let owner = *state.get_owner(&params.token_id)?;
+
+    if let Some(approved) = params.approved {
+        ensure!(owner != approved, ContractError::ApprovedIsOwner);
+    }
+    ensure!(sender == owner || state.is_operator(&sender, &owner), ContractError::Unauthorized);
+
+    state.approve(&params.approved, &params.token_id);
+
+    logger.log(&Event::Approval {
+        owner,
+        approved: params.approved,
+        token_id: params.token_id,
+    })?;
+
+    Ok(A::accept())
+}
+
+/// Enable or disable some address as an operator of the address of the sender.
+/// Logs an `ApproveForAll` event.
+///
+/// It rejects if:
+/// - It fails to parse the parameter.
+/// - The operator address is the same as the sender address.
+#[receive(
+    contract = "erc721",
+    name = "setApprovalForAll",
+    parameter = "ApproveForAllParams",
+    enable_logger
+)]
+fn contract_set_approval_for_all<A: HasActions>(
+    ctx: &impl HasReceiveContext,
+    logger: &mut impl HasLogger,
+    state: &mut State,
+) -> ContractResult<A> {
+    let params: ApproveForAllParams = ctx.parameter_cursor().get()?;
+
+    let sender = ctx.sender();
+
+    // No reason to be an operator yourself
+    ensure!(params.operator != sender, ContractError::OperatorIsSender);
+
+    state.approval_for_all(&sender, &params.operator, params.approved);
+
+    logger.log(&Event::ApprovalForAll {
+        owner:    sender,
+        operator: params.operator,
+        approved: params.approved,
+    })?;
+
+    Ok(A::accept())
+}
+
+/// Optional contract function called when a token is transferred to an instance
+/// of this contract. The parameter include a `data` field which can be used to
+/// implement some arbitrary functionality. In this example we choose not to use
+/// it, and define the function to forward any transfers to the owner of the
+/// contract instance.
+///
+/// It rejects if:
+/// - It fails to parse the parameter.
+/// - Sender is not a contract.
+/// - `safeTransferFrom` to instance owner rejects
+#[receive(contract = "erc721", name = "onERC721Received")]
+fn contract_on_erc721_received<A: HasActions>(
+    ctx: &impl HasReceiveContext,
+    _state: &mut State,
+) -> ContractResult<A> {
+    let params: OnERC721ReceivedParams = ctx.parameter_cursor().get()?;
+
+    let sender = if let Address::Contract(contract) = ctx.sender() {
+        contract
+    } else {
+        bail!(ContractError::ContractOnly)
+    };
+
+    let parameter = SafeTransferFromParams {
+        from:         Address::Contract(ctx.self_address()),
+        to:           Address::Account(ctx.invoker()),
+        token_id:     params.token_id,
+        receive_name: None,
+        data:         vec![],
+    };
+
+    let mut receive_name_string = params.contract_name.get_chain_name().clone();
```

Comment:
Shouldn't this be `contract_name` instead of `get_chain_name`? `chain_name` will have the `init_` prefix.

Reply:
You are right